test/starknet/EthereumSigAuth.test.ts

```diff
@@ -0,0 +1,295 @@
+import { hash } from 'starknet';
+import { SplitUint256, Choice } from '../shared/types';
+import { createExecutionHash, flatten2DArray } from '../shared/helpers';
+import { strToShortStringArr } from '@snapshot-labs/sx';
+import { expect } from 'chai';
+import { ethereumSigSetup } from '../shared/setup';
+import { StarknetContract, Account } from 'hardhat/types';
+import { ethers } from 'hardhat';
+import { domain, Propose, proposeTypes, Vote, voteTypes } from './helper/types';
+import { _TypedDataEncoder } from '@ethersproject/hash';
+import { TypedDataDomain, TypedDataField } from '@ethersproject/abstract-signer';
+import {
+  recoverPublicKey,
+  verifyMessage,
+  verifyTypedData,
+  toUtf8Bytes,
+  computePublicKey,
+  computeAddress,
+  keccak256,
+  hexConcat,
+  hexZeroPad,
+  serializeTransaction,
+} from 'ethers/lib/utils';
+import { type } from 'os';
+import { keccak } from 'ethereumjs-util';
+import { Signature } from 'ethers';
+
+const { getSelectorFromName } = hash;
+
+export const VITALIK_ADDRESS = BigInt('0xd8da6bf26964af9d7eed9e03e53415d37aa96045');
+export const AUTHENTICATE_METHOD = 'authenticate';
+export const PROPOSAL_METHOD = 'propose';
+export const VOTE_METHOD = 'vote';
+
+function getHash(
+  domain: TypedDataDomain,
+  types: Record<string, TypedDataField[]>,
+  message: Record<string, any>
+) {
+  const msgHash = _TypedDataEncoder.hash(domain, types, message);
+
+  // Stub code to generate and print the type hash
+  // const vote = "Vote(uint256 salt,bytes32 space,uint256 proposal,uint256 choice)";
+  // let s = Buffer.from(vote);
+  // let typeHash: string = keccak256(s);
+  // console.log("typeHash: ", typeHash);
+
+  return msgHash;
+}
+
+function prefixWithZeroes(s: string) {
+  // Remove prefix
+  if (s.startsWith('0x')) {
+    s = s.substring(2);
+  }
+
+  const numZeroes = 64 - s.length;
+  return '0x' + '0'.repeat(numZeroes) + s;
+}
+
+function hexPadRight(s: string) {
+  // Remove prefix
+  if (s.startsWith('0x')) {
+    s = s.substring(2);
+  }
+
+  // Odd length, need to prefix with a 0
+  if (s.length % 2 != 0) {
+    s = '0' + s;
+  }
+
+  const numZeroes = 64 - s.length;
+  return '0x' + s + '0'.repeat(numZeroes);
+}
+
+// Extracts and returns the `r, s, v` values from a `signature`.
+// `r`, `s` are SplitUint256, `v` is a BigInt
+function getRSVFromSig(sig: string) {
+  if (sig.startsWith('0x')) {
+    sig = sig.substring(2);
+  }
+  const r = SplitUint256.fromHex('0x' + sig.substring(0, 64));
+  const s = SplitUint256.fromHex('0x' + sig.substring(64, 64 * 2));
+  const v = BigInt('0x' + sig.substring(64 * 2));
+  return { r, s, v };
+}
+
+describe('Ethereum Sig Auth testing', () => {
```

Comment:
I refacotred all the tests to make them consistent and more readable. Would be good to adapt this to that, check space.test.ts for inspo

Reply:
done :D